### PR TITLE
Adalyn keyboard initial

### PR DESCRIPTION
**Pull request by tominabox1**
## Description

Initial support for Adalyn, an Alice-Like QAZ keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/adalyn/adalyn.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "adalyn.h"
```
Comment (drashna) [inline keyboards/adalyn/adalyn.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/adalyn/adalyn.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (drashna) [inline keyboards/adalyn/adalyn.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/adalyn/config.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#pragma once
```
Comment (drashna) [inline keyboards/adalyn/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/adalyn/keymaps/default/config.h [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#pragma once
```
Comment (drashna) [inline keyboards/adalyn/keymaps/default/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/adalyn/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/adalyn/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/adalyn/rules.mk [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+COMBO_ENABLE = yes
```
Comment (drashna) [inline keyboards/adalyn/rules.mk:23]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/adalyn/keymaps/vial/config.h [hunk 7]
```diff
@@ -0,0 +1,21 @@
+ /* Copyright TJ Campie
```
Comment (drashna) [inline keyboards/adalyn/keymaps/vial/config.h:1]: VIAL is not supported in the main repo, and the vial config and keymap should be removed.

File: keyboards/adalyn/keymaps/vial/keymap.c [hunk 8]
```diff
@@ -0,0 +1,49 @@
+ /* Copyright TJ Campie
```
Comment (drashna) [inline keyboards/adalyn/keymaps/vial/keymap.c:1]: VIAL is not supported in the main repo, and the vial config and keymap should be removed.

File: keyboards/adalyn/keymaps/vial/rules.mk [hunk 9]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
```
Comment (drashna) [inline keyboards/adalyn/keymaps/vial/rules.mk:1]: VIAL is not supported in the main repo, and the vial config and keymap should be removed.

File: keyboards/adalyn/keymaps/vial/vial.json [hunk 10]
```diff
@@ -0,0 +1,154 @@
+
```
Comment (drashna) [inline keyboards/adalyn/keymaps/vial/vial.json:1]: VIAL is not supported in the main repo, and the vial config and keymap should be removed.

File: keyboards/adalyn/rules.mk [hunk 11]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/adalyn/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/adalyn/rules.mk [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/adalyn/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/adalyn/rules.mk [hunk 13]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes     # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/adalyn/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
```

File: keyboards/adalyn/info.json [hunk 14]
```diff
@@ -0,0 +1,12 @@
+{
+  "keyboard_name": "adalyn",
+  "url": "https://github.com/MarvFPV/Adalyn",
+  "maintainer": "tominabox1",
+  "width": 10.25,
+  "height": 4,
```
Comment (fauxpark) [inline keyboards/adalyn/info.json:6]:
```suggestion
```

File: keyboards/adalyn/readme.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+# Adalyn
+
+Todo
```
Comment (fauxpark) [inline keyboards/adalyn/readme.md:3]:
```suggestion
![%KEYBOARD%](imgur.com image replace me!)

*A short description of the keyboard/project*

* Keyboard Maintainer: [%YOUR_NAME%](https://github.com/yourusername)
* Hardware Supported: *The PCBs, controllers supported*
* Hardware Availability: *Links to where you can find this hardware*

Make example for this keyboard (after setting up your build environment):

    make %KEYBOARD%:default

Flashing example for this keyboard:

    make %KEYBOARD%:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**